lib/templates.ts

```diff
@@ -325,6 +328,67 @@ export async function getAllTemplates(
   }
 }
 
+/**
+ * Get all form templates for a specific User.
+ * @returns An array of Form Records
+ */
+export async function getAllTemplatesForUser(
+  ability: UserAbility,
+  userID: string,
+  isPublished?: boolean
+): Promise<Array<FormRecord>> {
+  try {
+    checkPrivileges(ability, [{ action: "view", subject: "FormRecord" }]);
```

Comment:
Note to myself to check into further because I have a gut feeling this doesn't do what we may intend it to check for.

Reply:
Since we're allowing templates to be pulled for any user and not just the user identifed on the Ability we should also check that the user has the "View User" privilege.